File: java/json_control_chars.java
//Script takes control characters and outputs valid JSON
import org.json.simple.JSONObject;
import org.json.simple.JSONValue;

public class ControlCharactersToJson {
    public static void main(String[] args) {
        if (args.length == 0) {
            System.out.println("Usage: java ControlCharactersToJson <test_string>");
            System.exit(1);
        }

        String testString = args[0];
        String jsonString = JSONValue.toJSONString(testString);
        System.out.println(jsonString);
    }
}

File: golang/json_control_chars.go
//Script takes control characters and outputs valid JSON
package main

import (
	"encoding/json"
	"fmt"
	"os"
)

func main() {
	testString := os.Args[1]

	jsonString, _ := json.Marshal(testString)

	fmt.Println(string(jsonString))
}
